source/common/router/config_utility.cc

```diff
@@ -98,6 +99,38 @@ Optional<Http::Code> ConfigUtility::parseDirectResponseCode(const envoy::api::v2
   return Optional<Http::Code>();
 }
 
+std::string ConfigUtility::parseDirectResponseBody(const envoy::api::v2::Route& route) {
+  if (!route.has_direct_response() || !route.direct_response().has_body()) {
+    return EMPTY_STRING;
+  }
+  const auto& body = route.direct_response().body();
+  std::string filename = body.filename();
+  if (!filename.empty()) {
+    static const ssize_t kMaxFileSize = 4096;
+    if (!Filesystem::fileExists(filename)) {
+      throw EnvoyException(fmt::format("response body file {} does not exist", filename));
+    }
+    ssize_t size = Filesystem::fileSize(filename);
+    if (size < 0) {
+      throw EnvoyException(fmt::format("cannot determine size of response body file {}", filename));
+    }
+    if (size > kMaxFileSize) {
+      throw EnvoyException(fmt::format("response body file {} size is {} bytes; maximum is {}",
+                                       filename, kMaxFileSize));
+    }
+    return Filesystem::fileReadToEnd(filename);
+  }
+  std::string inline_bytes = body.inline_bytes();
+  if (!inline_bytes.empty()) {
+    return inline_bytes;
+  }
+  std::string inline_string = body.inline_string();
```

Comment:
nit: you don't really need the extra logic below here. At this point you can just return `inline_string`.